common-rest/src/main/java/org/uniprot/api/rest/service/query/processor/UniProtFieldQueryNodeProcessor.java

```diff
@@ -86,20 +86,21 @@ public CharSequence toQueryString(EscapeQuerySyntax escaper) {
                 return defaultSearchToQueryString(text);
             } else if (validWhiteListFields(field, text)) {
                 return field.toUpperCase() + "\\:" + text;
-            } else if (UNIPROTKB_ACCESSION_FIELD.equals(field)) {
+            } else if (UNIPROTKB_ACCESSION_FIELD.equals(field) && validFieldIgnoreCase(field)) {
                 return field + SOLR_FIELD_SEPARATOR + text.toUpperCase();
             } else if (validFieldIgnoreCase(field)) {
                 return field.toLowerCase() + SOLR_FIELD_SEPARATOR + text;
-            } else {
-                // is an alias => return the field itself
-                Optional<SearchFieldItem> searchFieldItemByAlias =
-                        searchFieldConfig.findSearchFieldItemByAlias(field);
-                if (searchFieldItemByAlias.isPresent()) {
-                    return searchFieldItemByAlias.get().getFieldName()
-                            + SOLR_FIELD_SEPARATOR
-                            + text;
-                }
+            } else if (UNIPROTKB_ACCESSION_FIELD.equals(field)) {
+                text = text.toUpperCase();
             }
+
+            // is an alias => return the field itself
```

Comment:
Do you think we could extract this alias logic to a method?

Reply:
i dnt think extracting to a different method would make sense here. but I have simplified the flow of the current method